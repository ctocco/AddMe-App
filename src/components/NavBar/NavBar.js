import React from "react";

const NavBar = () => {
  return (
    <div>
      <h1>"hello"</h1>
    </div>
  );
};

export default NavBar;
