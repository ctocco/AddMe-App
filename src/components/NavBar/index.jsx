import React from "react";
import { Link } from "@reach/router";
import styled from "styled-components";
import AdminPic from "../../assets/img/AdminPic.png";
import home from "../../assets/icons/home.svg";
import add_user from "../../assets/icons/add_user.svg";
import view_users from "../../assets/icons/view_users.svg";

const Nav = styled.nav`
  display: grid;
  background: #39555b;
  grid-template-columns: auto;
  grid-template-rows: 200px 200px 600px;
`;

const LogoContainer = styled.div`
  display: grid;
  grid-temlate-columns: 1fr;
  grid-template-rows: 150px auto;
  justify-content: center;
  text-align: center;
  align-content: center;
  color: white;
  grid-gap: 5px;
`;

const Circle = styled.div`
  display: flex;
  justify-content: center;
  background: #ddd;
  width: 150px;
  height: 150px;
  border-radius: 50%;
`;

const NavLink = styled.div`
  display: flex;
  color: white;
  align-items: center;
`;

const LinksContainer = styled.section`
  display: grid;
  grid-template-rows: 1fr 1fr 1fr;
  border-top: 1px solid white;
  padding: 20px;
  justify-content: center;
`;

const NavBar = () => {
  return (
    <Nav>
      <LogoContainer>
        <Circle>
          <img src={AdminPic} alt={"placeholder of a key"} />
        </Circle>
        <h1 data-testid="h1">Admin</h1>
      </LogoContainer>

      <LinksContainer>
        <Link to="/">
          <NavLink>
            <img src={home} alt={"icon of home button"} />
            <div>Home</div>
          </NavLink>
        </Link>
        <Link to="/viewusers">
          <NavLink>
            <img src={view_users} alt={"icon of view users button"} />
            <div>View Users</div>
          </NavLink>
        </Link>
        <Link to="/addusers">
          <NavLink>
            <img src={add_user} alt={"icon of add user button"} />
            <div>Add Users</div>
          </NavLink>
        </Link>
      </LinksContainer>

      <div>Rest</div>
    </Nav>
  );
};

export default NavBar;
