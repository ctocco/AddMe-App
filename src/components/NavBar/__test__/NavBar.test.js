import React from "react";
import ReactDOM from "react-dom";
import "@testing-library/jest-dom/extend-expect";
import {
  render,
  fireEvent,
  waitForElement,
  cleanup
} from "@testing-library/react";
import renderer from "react-test-renderer";

import NavBar from "../index";
//import "jest-dom/extend-expect";

afterEach(cleanup); // cleans up

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<NavBar />, div);
});
test("renders", () => {
  const { asFragment } = render(<NavBar text="Hello!" />);
  expect(asFragment()).toMatchSnapshot();
});

it("inserts text in H1", () => {
  const { getByTestId, getByText } = render(<NavBar text="Admin" />);
  expect(getByTestId("h1")).toHaveTextContent("Admin");
});

it("matches snapshot", () => {
  const tree = renderer.create(<h1>Admin</h1>).toJSON;
  expect(tree).toMatchSnapshot();
});
