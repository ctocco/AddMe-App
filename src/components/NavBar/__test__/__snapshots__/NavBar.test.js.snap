// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `[Function]`;

exports[`renders 1`] = `
<DocumentFragment>
  <nav
    class="sc-bdVaJa doDQFS"
  >
    <div
      class="sc-bwzfXH hVgNAy"
    >
      <div
        class="sc-htpNat hPucVN"
      >
        <img
          alt="placeholder of a key"
          src="AdminPic.png"
        />
      </div>
      <h1
        data-testid="h1"
      >
        Admin
      </h1>
    </div>
    <section
      class="sc-ifAKCX izJrHi"
    >
      <a
        aria-current="page"
        href="/"
      >
        <div
          class="sc-bxivhb iYyxbw"
        >
          <img
            alt="icon of home button"
            src="home.svg"
          />
          <div>
            Home
          </div>
        </div>
      </a>
      <a
        href="/viewusers"
      >
        <div
          class="sc-bxivhb iYyxbw"
        >
          <img
            alt="icon of view users button"
            src="view_users.svg"
          />
          <div>
            View Users
          </div>
        </div>
      </a>
      <a
        href="/addusers"
      >
        <div
          class="sc-bxivhb iYyxbw"
        >
          <img
            alt="icon of add user button"
            src="add_user.svg"
          />
          <div>
            Add Users
          </div>
        </div>
      </a>
    </section>
    <div>
      Rest
    </div>
  </nav>
</DocumentFragment>
`;
